pkg/controllers/binding/binding_controller.go

```diff
@@ -44,19 +45,25 @@ func (c *ResourceBindingController) Reconcile(ctx context.Context, req controlle
 
 	binding := &workv1alpha1.ResourceBinding{}
 	if err := c.Client.Get(context.TODO(), req.NamespacedName, binding); err != nil {
-		// The resource no longer exist, clean up derived Work objects.
+		// The resource no longer exist, in which case we stop processing.
 		if apierrors.IsNotFound(err) {
-			return helper.DeleteWorks(c.Client, labels.Set{
-				workv1alpha1.ResourceBindingNamespaceLabel: req.Namespace,
-				workv1alpha1.ResourceBindingNameLabel:      req.Name,
-			})
+			if apierrors.IsNotFound(err) {
+				return controllerruntime.Result{}, nil
+			}
```

Comment:
Typo? Why check again? I mean `apierrors.IsNotFound(err)`.

Reply:
Same as below.

Reply:
oh sorry it's my oversight